src/physics/utilities/solver_config.hpp

```diff
@@ -121,10 +121,43 @@ struct HypreBoomerAMGPrec {
   mfem::ParFiniteElementSpace* pfes = nullptr;
 };
 
+/**
+ * @brief Solver types supported by AMGX
+ */
+enum class AMGXSolver
+{
+  AMG,
+  PCGF,
+  CG,
+  PCG,
+  PBICGSTAB,
+  BICGSTAB,
+  FGMRES,
+  JACOBI_L1,
+  GS,
+  POLYNOMIAL,
+  KPZ_POLYNOMIAL,
+  BLOCK_JACOBI,
+  MULTICOLOR_GS,
+  MULTICOLOR_DILU
+};
+
 /**
  * @brief Stores the information required to configure a NVIDIA AMGX preconditioner
  */
 struct AMGXPrec {
```

Comment:
This goes beyond this PR, but we should append these option-type structs with `Option` so it's clearer what these things are doing.